upup/pkg/fi/cloudup/azure/azure_utils.go

```diff
@@ -40,3 +40,119 @@ func ZoneToAvailabilityZoneNumber(zone string) (string, error) {
 	}
 	return l[1], nil
 }
+
+// SubnetID contains the resource ID/names required to construct a subnet ID.
+type SubnetID struct {
+	SubscriptionID     string
+	ResourceGroupName  string
+	VirtualNetworkName string
+	SubnetName         string
+}
+
+// String returns the subnet ID in the path format.
+func (s *SubnetID) String() string {
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s",
+		s.SubscriptionID,
+		s.ResourceGroupName,
+		s.VirtualNetworkName,
+		s.SubnetName)
+}
+
+// ParseSubnetID parses a given subnet ID string and returns a SubnetID.
+func ParseSubnetID(s string) (*SubnetID, error) {
+	l := strings.Split(s, "/")
+	if len(l) != 11 {
+		return nil, fmt.Errorf("malformed format of subnet ID: %s, %d", s, len(l))
```

Comment:
Nit: I normally prefer to print with %q to highlight the "wrong" value, and I'm not sure that printing the length really adds any information (though I know it is handy during development!)

Reply:
It's just copy/paste from the original code, wanted to be easy diff.